### ISSUE TITLE
Virtual Component Groups filtered by input type

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#4452]
This task is automatically imported from the old Task Issue Board and it was originally created by Mateusz Czapliński.
Original issue is [here](https://www.pivotaltracker.com/story/show/182661634).
<hr />

## Why

**As** Enso IDE developer
**I want** to have access to Virtual Component Groups filtered by this argument and input node in new Hierarchical Action List
**So that** I can implement displaying them in Component List Panel View.

## Acceptance Criteria

```gherkin
Scenario: 
GIVEN I have test with mocked Engine's response for `executionContext/getComponentGroups`
WHEN I create searcher controller with non-empty this argument or input node
THEN I receive Virtual Groups in Action List filtered by the type of the this argument or input node
```

## Implementation Notes
 - Virtual Component Groups in the Component List should only include components with IDs present in the completions list returned by the Engine.
 - Virtual Component Groups where none of the components match IDs from completions should still be present in the Component List. This is to ensure that column placement of Virtual Component Groups when rendered is the same regardless of their contents being filtered by type. Skipping rendering empty Virtual Component Groups will be the responsibility of a View.

## Comments:
**Technical Design** review set to **pass** (Adam Obuchowicz - Jul 7, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-07-07):

**Progress:** Implemented and started testing. Having some issues running the Engine locally for final tests; asked DB for help. It should be finished by 2022-07-12.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/182661634). Finish testing and open a PR. Ask AO what next tasks to work on. (Enso Bot - Jul 7, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for yesterday (2022-07-13):

**Summary:** There is **7 days** delay in implementation of the [Virtual Component Groups filtered by input type (#182661634) task](https://www.pivotaltracker.com/story/show/182661634).
It **will cause 1 day delay** for the delivery of this weekly plan.

(3 days delay receiving code-review +1 day extra work +3 days of weekend/vacation)

**Delay Cause:** First, receiving a code-review was delayed by 3 days (presumably due to other, higher priority tasks of reviewers). After review, the task will need 1 day of extra work (resolving non-trivial conflicts with PR 3530 + addressing code-review comments). In addition to that, delay will overlap 3 days of weekend/vacation. (Enso Bot - Jul 14, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-07-13):

**Progress:** Debugged unexpected issues after merging with PR 3530 (which is in progress of being merged into develop). Found the root cause to be conflicts in logic between the two PRs. Started working on a fix. It should be finished by 2022-07-19.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/182661634). Finish the fix and update the PR. Re-request review. If time permits, switch to #182584326 which should be unblocked by merging PR 3530. Re-test with new develop and open a PR if time permits. (Enso Bot - Jul 14, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-07-14):

**Progress:** Resolved issues caused by merge with PR 3530. Updated the PR and re-requested review from AO. After Approved, requested review from WD. It should be finished by 2022-07-19. (Enso Bot - Jul 14, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for yesterday (2022-07-20):

**Summary:** There is **1 day** delay in implementation of the [Virtual Component Groups filtered by input type (#182661634) task](https://www.pivotaltracker.com/story/show/182661634).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Address a big number of naming & documentation code-review comments. (Enso Bot - Jul 21, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-07-20):

**Progress:** Addressed a big number of naming & documentation code-review comments and re-requested review. It should be finished by 2022-07-20. (Enso Bot - Jul 21, 2022)
<hr />
**Acceptance/QA** review set to **pass** (Ilya Bogdanov - Jul 22, 2022)
<hr />
